zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -1955,6 +1960,7 @@ public void checkACL(ServerCnxn cnxn, List<ACL> acl, int perm, List<Id> ids, Str
                 ServerAuthenticationProvider ap = ProviderRegistry.getServerProvider(id.getScheme());
                 if (ap != null) {
                     for (Id authId : ids) {
+                        LOG.debug("Iterating IDs : {}", authId);
                         if (authId.getScheme().equals(id.getScheme())
                             && ap.matches(
                                 new ServerAuthenticationProvider.ServerObjs(this, cnxn),
```

Comment:
Is there a reason we're not logging the contents of these objects?

Reply:
Sorry i didn't get you? we are logging it inside matches method right? still i added few logs here let me know if i miss something here :)

Reply:
Gotcha - inside matches works!